### ISSUE TITLE
My server is crashing because of an incompatibility with the Roughly enough Items mod.

### ISSUE THREAD

**Issue by Everthon-Silva** [Salandora/SophisticatedBackpacks#5]
My server is crashing because of an incompatibility with the Roughly enough Items mod
version: 1.19.2
fabric: 0.15.10
REI: 9.2.710

**Comment by Salandora** [comment]
Need a crash log for that

**Comment by Everthon-Silva** [comment]
```
[14:00:21] [main/INFO]: Loading Minecraft 1.19.2 with Fabric Loader 0.15.10
[14:00:21] [main/INFO]: Loading 66 mods:
	- architectury 6.6.92
	- cloth-config 8.3.115
	   \-- cloth-basic-math 0.6.1
	- fabric-api 0.77.0+1.19.2
	   |-- fabric-api-base 0.4.15+8f4e8eb390
	   |-- fabric-api-lookup-api-v1 1.6.14+93d8cb8290
	   |-- fabric-biome-api-v1 9.1.1+16f1e31390
	   |-- fabric-block-api-v1 1.0.2+e415d50e90
	   |-- fabric-blockrenderlayer-v1 1.1.25+cafc6e8e90
	   |-- fabric-client-tags-api-v1 1.0.5+b35fea8390
	   |-- fabric-command-api-v1 1.2.16+f71b366f90
	   |-- fabric-command-api-v2 2.2.1+413cbbc790
	   |-- fabric-commands-v0 0.2.33+df3654b390
	   |-- fabric-containers-v0 0.1.42+df3654b390
	   |-- fabric-content-registries-v0 3.5.2+7c6cd14d90
	   |-- fabric-convention-tags-v1 1.3.0+4bc6e26290
	   |-- fabric-crash-report-info-v1 0.2.8+aeb40ebe90
	   |-- fabric-data-generation-api-v1 5.3.9+413cbbc790
	   |-- fabric-dimensions-v1 2.1.35+0d0f210290
	   |-- fabric-entity-events-v1 1.5.4+9244241690
	   |-- fabric-events-interaction-v0 0.4.34+562bff6e90
	   |-- fabric-events-lifecycle-v0 0.2.36+df3654b390
	   |-- fabric-game-rule-api-v1 1.0.24+b6b6abb490
	   |-- fabric-item-api-v1 1.6.6+b7d1888890
	   |-- fabric-item-groups-v0 0.3.39+9244241690
	   |-- fabric-key-binding-api-v1 1.0.25+5c4fce2890
	   |-- fabric-keybindings-v0 0.2.23+df3654b390
	   |-- fabric-lifecycle-events-v1 2.2.4+1b46dc7890
	   |-- fabric-loot-api-v2 1.1.13+83a8659290
	   |-- fabric-loot-tables-v1 1.1.16+9e7660c690
	   |-- fabric-message-api-v1 5.0.7+93d8cb8290
	   |-- fabric-mining-level-api-v1 2.1.24+33fbc73890
	   |-- fabric-models-v0 0.3.21+c6af733c90
	   |-- fabric-networking-api-v1 1.2.12+def3f86d90
	   |-- fabric-networking-v0 0.3.29+df3654b390
	   |-- fabric-object-builder-api-v1 4.2.2+d8ef690890
	   |-- fabric-particles-v1 1.1.0+ee641e7390
	   |-- fabric-recipe-api-v1 1.0.2+413cbbc790
	   |-- fabric-registry-sync-v0 0.9.33+9244241690
	   |-- fabric-renderer-api-v1 1.2.1+1adbf27790
	   |-- fabric-renderer-indigo 0.8.0+1adbf27790
	   |-- fabric-renderer-registries-v1 3.2.25+df3654b390
	   |-- fabric-rendering-data-attachment-v1 0.3.19+6e0787e690
	   |-- fabric-rendering-fluids-v1 3.0.11+4d0d570390
	   |-- fabric-rendering-v0 1.1.28+df3654b390
	   |-- fabric-rendering-v1 1.13.0+526f2c6790
	   |-- fabric-resource-conditions-api-v1 2.1.2+aae9039d90
	   |-- fabric-resource-loader-v0 0.8.4+edbdcddb90
	   |-- fabric-screen-api-v1 1.0.32+4d0d570390
	   |-- fabric-screen-handler-api-v1 1.3.8+1cc24b1b90
	   |-- fabric-sound-api-v1 1.0.2+c4f28df590
	   |-- fabric-textures-v0 1.0.24+aeb40ebe90
	   |-- fabric-transfer-api-v1 2.1.6+413cbbc790
	   \-- fabric-transitive-access-wideners-v1 1.3.3+08b73de490
	- fabricloader 0.15.10
	   \-- mixinextras 0.3.5
	- java 17
	- minecraft 1.19.2
	- roughlyenoughitems 9.2.710
	   \-- error_notifier 1.0.9
	- sophisticatedbackpacks 3.19.4+mc1.19.2-SNAPSHOT-build.12
	   |-- cardinal-components-base 5.0.2
	   |-- cardinal-components-entity 5.0.2
	   |-- forgeconfigapiport 4.2.11
	   |    |-- com_electronwill_night-config_core 3.6.5
	   |    \-- com_electronwill_night-config_toml 3.6.5
	   |-- porting_lib_common 2.1.1282+1.19.2
	   |-- porting_lib_extensions 2.1.1282+1.19.2
	   |    |-- porting_lib_accessors 2.1.1282+1.19.2
	   |    |-- porting_lib_attributes 2.1.1282+1.19.2
	   |    \-- porting_lib_constants 2.1.1282+1.19.2
	   |-- porting_lib_fake_players 2.1.1282+1.19.2
	   |-- porting_lib_loot 2.1.1282+1.19.2
	   |    \-- porting_lib_lazy_registration 2.1.1282+1.19.2
	   |-- porting_lib_model_loader 2.1.1282+1.19.2
	   |-- porting_lib_networking 2.1.1282+1.19.2
	   |-- porting_lib_transfer 2.1.1282+1.19.2
	   \-- team_reborn_energy 2.3.0
	- sophisticatedcore 0.5.108+mc1.19.2-SNAPSHOT-build.12
[14:00:21] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/home/container/libraries/net/fabricmc/sponge-mixin/0.13.3+mixin.0.8.5/sponge-mixin-0.13.3+mixin.0.8.5.jar Service=Knot/Fabric Env=SERVER
[14:00:21] [main/INFO]: Compatibility level set to JAVA_16
[14:00:21] [main/INFO]: Compatibility level set to JAVA_17
[14:00:22] [main/INFO]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.3.5).
[14:00:24] [main/INFO]: Building unoptimized datafixer
[14:00:25] [main/INFO]: Registering C2S receiver with id roughlyenoughitems:request_tags_c2s
[14:00:25] [main/INFO]: [REI] Registered plugin provider DefaultPlugin [roughlyenoughitems] for REIPlugin
[14:00:25] [main/INFO]: [REI] Registered plugin provider DefaultRuntimePlugin [roughlyenoughitems] for REIPlugin
[14:00:25] [main/INFO]: [REI] Registered plugin provider REICompat [sophisticatedbackpacks] for REIPlugin
[14:00:25] [main/INFO]: [REI] Registered plugin provider DefaultPlugin [roughlyenoughitems] for REIServerPlugin
[14:00:25] [main/INFO]: [REI] Registered plugin provider DefaultRuntimePlugin [roughlyenoughitems] for REIServerPlugin
[14:00:25] [main/INFO]: [REI] Registered plugin provider REICompat [sophisticatedbackpacks] for REIServerPlugin
[14:00:25] [main/INFO]: [REI] Registered plugin provider FabricFluidAPISupportPlugin for REIServerPlugin
[14:00:25] [main/INFO]: [REI] Registered plugin provider FabricFluidAPISupportPlugin for REIPlugin
[14:00:25] [main/INFO]: Registering C2S receiver with id roughlyenoughitems:delete_item
[14:00:25] [main/INFO]: Registering C2S receiver with id roughlyenoughitems:create_item
[14:00:25] [main/INFO]: Registering C2S receiver with id roughlyenoughitems:create_item_grab
[14:00:25] [main/INFO]: Registering C2S receiver with id roughlyenoughitems:create_item_hotbar
[14:00:25] [main/INFO]: Registering C2S receiver with id roughlyenoughitems:move_items
[14:00:25] [main/INFO]: Registering C2S receiver with id roughlyenoughitems:move_items_new
[14:00:25] [main/WARN]: Configuration file ./config/sophisticatedbackpacks-client.toml is not correct. Correcting
[14:00:25] [main/WARN]: Incorrect key common was corrected from null to its default, SimpleCommentedConfig:{}. 
[14:00:25] [main/WARN]: Incorrect key common.chestLootEnabled was corrected from null to its default, true. 
[14:00:25] [main/ERROR]: No data fixer registered for 
[14:00:25] [main/WARN]: Configuration file ./config/sophisticatedcore-client.toml is not correct. Correcting
[14:00:25] [main/WARN]: Incorrect key client was corrected from null to its default, SimpleCommentedConfig:{}. 
[14:00:25] [main/WARN]: Incorrect key client.sortButtonsPosition was corrected from null to its default, TITLE_LINE_RIGHT. 
[14:00:25] [main/WARN]: Incorrect key client.playButtonSound was corrected from null to its default, true. 
[14:00:25] [main/ERROR]: Failed to start the minecraft server
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'sophisticatedcore'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:63) ~[fabric-loader-0.15.10.jar:?]
	at net.minecraft.server.Main.main(Main.java:101) [server-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) [fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.15.10.jar:?]
	at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) [server.jar:1.0.1]
Caused by: java.lang.BootstrapMethodError: java.lang.RuntimeException: Cannot load class me.shedaniel.rei.api.client.plugins.REIClientPlugin in environment type SERVER
	at net.p3pp3rf1y.sophisticatedcore.init.ModCompat.lambda$static$2(ModCompat.java:25) ~[sophisticatedcore-0.5.108+mc1.19.2-SNAPSHOT-build.12.jar:?]
	at net.p3pp3rf1y.sophisticatedcore.init.ModCompat.initCompats(ModCompat.java:34) ~[sophisticatedcore-0.5.108+mc1.19.2-SNAPSHOT-build.12.jar:?]
	at net.p3pp3rf1y.sophisticatedcore.SophisticatedCore.onInitialize(SophisticatedCore.java:44) ~[sophisticatedcore-0.5.108+mc1.19.2-SNAPSHOT-build.12.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384) ~[fabric-loader-0.15.10.jar:?]
	... 7 more
Caused by: java.lang.RuntimeException: Cannot load class me.shedaniel.rei.api.client.plugins.REIClientPlugin in environment type SERVER
	at net.fabricmc.loader.impl.transformer.FabricTransformer.transform(FabricTransformer.java:59) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPreMixinClassByteArray(KnotClassDelegate.java:462) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:415) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.15.10.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1017) ~[?:?]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:160) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:355) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.15.10.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at net.p3pp3rf1y.sophisticatedcore.init.ModCompat.lambda$static$2(ModCompat.java:25) ~[sophisticatedcore-0.5.108+mc1.19.2-SNAPSHOT-build.12.jar:?]
	at net.p3pp3rf1y.sophisticatedcore.init.ModCompat.initCompats(ModCompat.java:34) ~[sophisticatedcore-0.5.108+mc1.19.2-SNAPSHOT-build.12.jar:?]
	at net.p3pp3rf1y.sophisticatedcore.SophisticatedCore.onInitialize(SophisticatedCore.java:44) ~[sophisticatedcore-0.5.108+mc1.19.2-SNAPSHOT-build.12.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384) ~[fabric-loader-0.15.10.jar:?]
	... 7 more
```
and REI log:
```
[14:00:25] [main/INFO] ========================================
[14:00:25] [main/INFO] Minecraft: 1.19.2
[14:00:25] [main/INFO] Side: server
[14:00:25] [main/INFO] Development: false
[14:00:25] [main/INFO] Version: 9.2.710
[14:00:25] [main/INFO] Loader:
[14:00:25] [main/INFO] - Fabric: 0.15.10
[14:00:25] [main/INFO] - Fabric API: 0.77.0+1.19.2
[14:00:25] [main/INFO] Dependencies:
[14:00:25] [main/INFO] - Cloth Config: 8.3.115
[14:00:25] [main/INFO] - Architectury: 6.6.92
[14:00:25] [main/INFO] Mixin: 0.8.5
[14:00:25] [main/INFO] OS: Linux 5.15.0-97-generic (amd64)
[14:00:25] [main/INFO] Java: 17.0.9 (Amazon.com Inc.)
[14:00:25] [main/INFO] ========================================
[14:00:25] [main/INFO] Mods:
[14:00:25] [main/INFO] - architectury: 6.6.92
[14:00:25] [main/INFO] - cardinal-components-base: 5.0.2
[14:00:25] [main/INFO] - cardinal-components-entity: 5.0.2
[14:00:25] [main/INFO] - cloth-basic-math: 0.6.1
[14:00:25] [main/INFO] - cloth-config: 8.3.115
[14:00:25] [main/INFO] - cloth-config: 8.3.115
[14:00:25] [main/INFO] - com_electronwill_night-config_core: 3.6.5
[14:00:25] [main/INFO] - com_electronwill_night-config_toml: 3.6.5
[14:00:25] [main/INFO] - error_notifier: 1.0.9
[14:00:25] [main/INFO] - fabric-api: 0.77.0+1.19.2
[14:00:25] [main/INFO] - fabric-api: 0.77.0+1.19.2
[14:00:25] [main/INFO] - fabric-api-base: 0.4.15+8f4e8eb390
[14:00:25] [main/INFO] - fabric-api-lookup-api-v1: 1.6.14+93d8cb8290
[14:00:25] [main/INFO] - fabric-biome-api-v1: 9.1.1+16f1e31390
[14:00:25] [main/INFO] - fabric-block-api-v1: 1.0.2+e415d50e90
[14:00:25] [main/INFO] - fabric-command-api-v1: 1.2.16+f71b366f90
[14:00:25] [main/INFO] - fabric-command-api-v2: 2.2.1+413cbbc790
[14:00:25] [main/INFO] - fabric-commands-v0: 0.2.33+df3654b390
[14:00:25] [main/INFO] - fabric-containers-v0: 0.1.42+df3654b390
[14:00:25] [main/INFO] - fabric-content-registries-v0: 3.5.2+7c6cd14d90
[14:00:25] [main/INFO] - fabric-convention-tags-v1: 1.3.0+4bc6e26290
[14:00:25] [main/INFO] - fabric-crash-report-info-v1: 0.2.8+aeb40ebe90
[14:00:25] [main/INFO] - fabric-data-generation-api-v1: 5.3.9+413cbbc790
[14:00:25] [main/INFO] - fabric-dimensions-v1: 2.1.35+0d0f210290
[14:00:25] [main/INFO] - fabric-entity-events-v1: 1.5.4+9244241690
[14:00:25] [main/INFO] - fabric-events-interaction-v0: 0.4.34+562bff6e90
[14:00:25] [main/INFO] - fabric-events-lifecycle-v0: 0.2.36+df3654b390
[14:00:25] [main/INFO] - fabric-game-rule-api-v1: 1.0.24+b6b6abb490
[14:00:25] [main/INFO] - fabric-item-api-v1: 1.6.6+b7d1888890
[14:00:25] [main/INFO] - fabric-item-groups-v0: 0.3.39+9244241690
[14:00:25] [main/INFO] - fabric-lifecycle-events-v1: 2.2.4+1b46dc7890
[14:00:25] [main/INFO] - fabric-loot-api-v2: 1.1.13+83a8659290
[14:00:25] [main/INFO] - fabric-loot-tables-v1: 1.1.16+9e7660c690
[14:00:25] [main/INFO] - fabric-message-api-v1: 5.0.7+93d8cb8290
[14:00:25] [main/INFO] - fabric-mining-level-api-v1: 2.1.24+33fbc73890
[14:00:25] [main/INFO] - fabric-networking-api-v1: 1.2.12+def3f86d90
[14:00:25] [main/INFO] - fabric-networking-v0: 0.3.29+df3654b390
[14:00:25] [main/INFO] - fabric-object-builder-api-v1: 4.2.2+d8ef690890
[14:00:25] [main/INFO] - fabric-particles-v1: 1.1.0+ee641e7390
[14:00:25] [main/INFO] - fabric-recipe-api-v1: 1.0.2+413cbbc790
[14:00:25] [main/INFO] - fabric-registry-sync-v0: 0.9.33+9244241690
[14:00:25] [main/INFO] - fabric-rendering-data-attachment-v1: 0.3.19+6e0787e690
[14:00:25] [main/INFO] - fabric-rendering-fluids-v1: 3.0.11+4d0d570390
[14:00:25] [main/INFO] - fabric-resource-conditions-api-v1: 2.1.2+aae9039d90
[14:00:25] [main/INFO] - fabric-resource-loader-v0: 0.8.4+edbdcddb90
[14:00:25] [main/INFO] - fabric-screen-handler-api-v1: 1.3.8+1cc24b1b90
[14:00:25] [main/INFO] - fabric-transfer-api-v1: 2.1.6+413cbbc790
[14:00:25] [main/INFO] - fabric-transitive-access-wideners-v1: 1.3.3+08b73de490
[14:00:25] [main/INFO] - fabricloader: 0.15.10
[14:00:25] [main/INFO] - forgeconfigapiport: 4.2.11
[14:00:25] [main/INFO] - java: 17
[14:00:25] [main/INFO] - minecraft: 1.19.2
[14:00:25] [main/INFO] - mixinextras: 0.3.5
[14:00:25] [main/INFO] - porting_lib_accessors: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_attributes: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_common: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_constants: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_extensions: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_fake_players: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_lazy_registration: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_loot: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_model_loader: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_networking: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - porting_lib_transfer: 2.1.1282+1.19.2
[14:00:25] [main/INFO] - roughlyenoughitems: 9.2.710
[14:00:25] [main/INFO] - sophisticatedbackpacks: 3.19.4+mc1.19.2-SNAPSHOT-build.12
[14:00:25] [main/INFO] - sophisticatedcore: 0.5.108+mc1.19.2-SNAPSHOT-build.12
[14:00:25] [main/INFO] - team_reborn_energy: 2.3.0
[14:00:25] [main/INFO] ========================================
[14:00:25] [main/INFO] [REI] Registered plugin provider DefaultPlugin [roughlyenoughitems] for REIPlugin
[14:00:25] [main/INFO] [REI] Registered plugin provider DefaultRuntimePlugin [roughlyenoughitems] for REIPlugin
[14:00:25] [main/INFO] [REI] Registered plugin provider REICompat [sophisticatedbackpacks] for REIPlugin
[14:00:25] [main/INFO] [REI] Registered plugin provider DefaultPlugin [roughlyenoughitems] for REIServerPlugin
[14:00:25] [main/INFO] [REI] Registered plugin provider DefaultRuntimePlugin [roughlyenoughitems] for REIServerPlugin
[14:00:25] [main/INFO] [REI] Registered plugin provider REICompat [sophisticatedbackpacks] for REIServerPlugin
[14:00:25] [main/INFO] [REI] Registered plugin provider FabricFluidAPISupportPlugin for REIServerPlugin
[14:00:25] [main/INFO] [REI] Registered plugin provider FabricFluidAPISupportPlugin for REIPlugin

```

**Comment by Salandora** [comment]
Thank you.
this is actually a bug I stumbled across myself recently and fixed it in: 5d9290e48edbf81f5db7ff04da8eaa83078e9328
Therefore will be fixed in the next release.
Until then I see 2 options: 
a) don't use REI (will leave you with the issue mentioned down below)
or 
b) grab the compiled versions from here: https://github.com/Salandora/SophisticatedMods/actions/runs/8856818860

Also just so you know, you'll have disconnect issues when using the smelter upgrade, this is also fixed in the aforementioned compiled version.

**Comment by Everthon-Silva** [comment]
Thank you sm!
